### PR TITLE
Openapi > SwaggerUI compatibility...

### PR DESCRIPTION
**Pull request by flyusfly**
This PR allows to use /openapi schema in SwaggerUI.

- [ ] Sort open api schema
- [x] Rename DiffEndpoint in Diff
- [ ] Add Redirect model to openapi json schema
- [ ] Add generic responses in openapi schema
- [ ] ModelEndpoint.get_collection() - Add a 401 response
- [ ] ModelEndpoint.get_resource() - Normalize & Add responses
- [ ] ModelEndpoint.post_resource() - Normalize & Add responses
- [ ] ModelEndpoint.post() - Normalize & Add responses
- [ ] ModelEndpoint.patch() - Normalize & Add responses
- [ ] ModelEndpoint.put() - Normalize & Add responses
- [ ] ModelEndpoint.delete() - Normalize & Add responses
- [ ] VersionedModelEndpoint.get_versions() - Normalize & Add responses
- [x] Rename VersionedModelEnpoint in VersionedModelEndpoint
- [x] Schema.get_responder_summary return resource name
- [ ] VersionedModelEndpoint.get_version() - Normalize & Add responses
- [ ] VersionedModelEndpoint.post_version() - Normalize & Add responses
- [ ] VersionedModelEndpoint.get_redirects() - Normalize & Add responses
- [ ] VersionedModelEndpoint.put_delete_redirects() > Split put and delete redirects & Normalize & Add responses
- [ ] Diff.get_collection > Normalize & Add responses
- [ ] 410 & 422 responses > General definitions & update responses

### CODE REVIEW

File: ban/core/versioning.py [hunk 1]
```diff
@@ -272,6 +272,13 @@ def serialize(self, *args):
 
 class Redirect(db.Model):
 
+    __openapi__ = """
+        properties:
+            identifier_value:
```
Comment (yohanboniface) [inline ban/core/versioning.py]: Humm, what `identifier_value`?

File: ban/http/api.py [hunk 2]
```diff
@@ -173,11 +173,20 @@ def get_resource(self, identifier):
 
         parameters:
             - $ref: '#/parameters/identifier'
+              name: identifier
```
Comment (yohanboniface) [inline ban/http/api.py]: Isn't `identifier` already referenced on the line above?

File: ban/http/api.py [hunk 3]
```diff
@@ -203,15 +203,33 @@ def get_resource(self, identifier):
...
+        """Post {resource} with 'identifier'.
 
         parameters:
             - $ref: '#/parameters/identifier'
+              name: identifier
+              in: path
+              type: string
+              required: true
+              description: {resource} identifier
+            - name: body
```
Comment (yohanboniface) [inline ban/http/api.py]: How is `body` a parameter?

File: ban/http/api.py [hunk 4]
```diff
@@ -429,7 +429,7 @@ def delete(self, identifier):
         return {'resource_id': identifier}
 
 
-class VersionedModelEnpoint(ModelEndpoint):
+class VersionedModelEndpoint(ModelEndpoint):
```
Comment (yohanboniface) [inline ban/http/api.py]: Nice catch :)

File: ban/http/schema.py [hunk 5]
```diff
@@ -68,7 +68,7 @@ def __init__(self, *args, **kwargs):
 
     def get_responder_summary(self, responder, resource):
         return (responder.__doc__ or '').split('\n\n')[0].format(
-            resource=resource.__class__.__name__)
```
Comment (yohanboniface) [inline ban/http/schema.py]: Good catch :)

File: ban/http/api.py (hunk 2 above)
Comment (flyusfly) [inline ban/http/api.py]: Flex needs that, in this case, a name must be defined.
If not, this is what error message shows :
```self = defaultdict(<class 'flex.exceptions.ErrorList'>, {'paths': [defaultdict(<class 'flex.exceptions.ErrorList'>, {'/group/..., {'required': [defaultdict(<class 'flex.exceptions.ErrorList'>, {'name': ['This value is required']})]})})]})]})]})]})```

File: ban/http/api.py (hunk 3 above)
Comment (flyusfly) [inline ban/http/api.py]: In this case, 'body' represents json text to give to create a post_resource.
May be we can rename this parameter fom 'body' to 'json'?

File: ban/core/versioning.py (hunk 1 above)
Comment (flyusfly) [inline ban/core/versioning.py]: When you make a redirect, you have to give :
   . identifier name
   . ':'
   . value

So, what comes to me to define it was 'identifier_value'.
Another idea ?

**Comment by flyusfly** [comment]
Let's split this PR into small ones...

**Pull request closed without merging**